### DISCUSSION TITLE
WasmEdge's Internal flow

### DISCUSSION THREAD

**Discussion by lollopelle01** [WasmEdge/WasmEdge#2987] [Q&A]
Hello everyone,

I'm currently working on my undergraduate thesis focused on WasmEdge, and I'm particularly intrigued by its internal workings. I came across the diagram ( https://github.com/wasmedge/docs/blob/main/docs/contribute/internal.md )  that illustrates the internal flow, but upon examining the source code, I noticed that some components do not seem to exist or are implemented differently from what is depicted.

For instance, I couldn't find an 'Instantiator' but there is an 'Executor' that seems to handle both the instantiation and execution of modules. I also couldn't locate an 'Interpreter Engine' or a 'WasmEdge Engine.' Without specific documentation on these aspects, I'm finding it challenging to establish a clear connection between the concepts and their actual implementations.

Do any of you have more detailed documentation, or could you provide assistance in reconstructing a more accurate 'Internal Flow'? Any insights or guidance would be greatly appreciated.

Thank you.

**Comment by dannypsnl** [comment]
Hi, I think the diagram is abstract and will always be, so for instance you can try to lookup with `rg instanti` to seek similar wording. Let every detail up to date is a heavy work, and unreasonable.
